qadence2_platforms/backends/_base_analog/functions.py

```diff
@@ -0,0 +1,212 @@
+            num_qubits (int): the number of qubits to create the qutip object to
+            expr (InputType): the input expression. Any qadence2-expressions expression
+                compatible object, with the same methods
+
+        Returns:
+            A QuTiP object with the Hilbert space compatible with `num_qubits`
+        """
+
+        subspace: set = cast(Support, expr.subspace).subspace
+        super_space: set = set(range(num_qubits))
+
+        if super_space.issuperset(subspace):
+            native_ops: list[qutip.Qobj] = []
+            arg_subspace: set = cast(Support, expr.args[1]).subspace
+
+            for k in range(num_qubits):
+                if k in arg_subspace:
+                    sub_num_qubits: int = cast(int, expr.args[1])
+                    arg: InputType = cast(InputType, expr.args[0])
+                    op: qutip.Qobj = cls._get_op(sub_num_qubits, arg)
+                    native_ops.append(op)
+                else:
+                    native_ops.append(cls.operators_mapping["I"])
+
+            return qutip.tensor(*native_ops)
+
+        raise ValueError(
+            f"subspace of the object ({subspace}) is bigger than the "
+            f"sequence space ({super_space})"
+        )
+
+    @classmethod
+    def _get_op(cls, num_qubits: int, op: InputType) -> qutip.Qobj | None:
+        """
+        Convert an expression into a native QuTiP object. A simple symbol,
+        a quantum operator, and an operation (addition, multiplication or
+        kron tensor) are valid objects.
+
+        Args:
+            num_qubits (int): the number of qubits to create the qutip object to
+            op (InputType): the input expression. Any qadence2-expressions expression
+                compatible object, with the same methods
+
+        Returns:
+            A QuTiP object with the Hilbert space compatible with `num_qubits`
+        """
+
+        if op.is_symbol is True:
+            symbol: str = cast(str, op.args[0])
+            return cls.operators_mapping[symbol]
+
+        op_arg: qutip.Qobj
+
+        if op.is_quantum_operator is True:
+            sub_num_qubits: int = cast(Support, op.args[1]).max_index + 1
+
+            if sub_num_qubits < num_qubits:
+                op_arg = cls._compl_tensor_op(num_qubits, op)
+
+            else:
+                arg: InputType = cast(InputType, op.args[0])
+                op_arg = cls._get_op(num_qubits, arg)
+
+            return op_arg
+
+        ops: list[qutip.Qobj] = []
+        args: Iterable = cast(Iterable, op.args)
+
+        if op.is_addition is True:
+
+            for arg in args:
+                ops.append(cls._arith_tensor_op(num_qubits, arg))
+
+            return reduce(lambda a, b: a + b, ops)
+
+        if op.is_multiplication is True:
+
+            for arg in args:
+                ops.append(cls._arith_tensor_op(num_qubits, arg))
+
+            return reduce(lambda a, b: a * b, ops)
+
+        if op.is_kronecker_product is True:
```

Comment:
Maybe make these clauses mutually exclusive ?

Reply:
no need, the previous function already returns something